test/libsolidity/ASTPropertyTests/simpleLoopOptimization.sol

```diff
@@ -0,0 +1,30 @@
+contract C {
```

Comment:
Nice! This very much looks like I'd have imagined this already - should split out the other stub test cases not connected to the counter loops into a separate file, but looks good!